treematcher.py

```diff
@@ -1,286 +1,178 @@
-from itertools import permutations
-from ete3 import NCBITaxa
-import six
+#system modules
 import re
-import ast
+from itertools import permutations
 
+# third party modules
+import ast
+import six
 from ete3 import PhyloTree, Tree, NCBITaxa
 
+# internal modules
+# ...
 
-class TreePattern(Tree):
-    """
-        Class for presenting a tree of constrained patterns. A way to represent patterns to search for within trees.
-        :param args: Pattern to be searched.
-        :param kargs: Custom functions and arguments passed to Tree Class.
-    """
-    _syntax_tuples = [
-        ("@", "__target"),
-        ("__target.size", "len(__target.get_leaves())"),
-        ("__target.leaves", "[node.name for node in __target.get_leaves()]"),
-        ("__target.contains_species", "[n.species for n in __target.traverse()]"),
-    ]
-
-    _syntax_tuples_with_cache = [
-        ("@", "__target"),
-        ("__target.size", "len(temp_leaf_cache[__target])"),
-        ("__target.contains_species", "[n.species for n in temp_leaf_cache[__target]]"),
-        ("__target.leaves", "[n.name for n in temp_leaf_cache[__target]]"),
-
-    ]
+class TreePatternCache(object):
```

Comment:
This is now use to generate a cache object for a tree. the object can be passed to TreePattern.find_match to be used in all functions